main.tf

```diff
@@ -56,52 +100,198 @@ resource "aws_launch_configuration" "this" {
+    }
+  }
+
+  dynamic "metadata_options" {
+    for_each = var.metadata_options != null ? [var.metadata_options] : []
+    content {
+      http_endpoint               = lookup(metadata_options.value, "http_endpoint", null)
+      http_tokens                 = lookup(metadata_options.value, "http_tokens", null)
+      http_put_response_hop_limit = lookup(metadata_options.value, "http_put_response_hop_limit", null)
+    }
+  }
+
+  dynamic "monitoring" {
+    for_each = var.enable_monitoring != null ? [1] : []
+    content {
+      enabled = var.enable_monitoring
+    }
+  }
+
+  dynamic "network_interfaces" {
+    for_each = var.network_interfaces
+    content {
+      associate_carrier_ip_address = lookup(network_interfaces.value, "associate_carrier_ip_address", null)
+      associate_public_ip_address  = lookup(network_interfaces.value, "associate_public_ip_address", null)
+      delete_on_termination        = lookup(network_interfaces.value, "delete_on_termination", null)
+      description                  = lookup(network_interfaces.value, "description", null)
+      device_index                 = lookup(network_interfaces.value, "device_index", null)
+      ipv4_addresses               = lookup(network_interfaces.value, "ipv4_addresses", null) != null ? network_interfaces.value.ipv4_addresses : []
+      ipv4_address_count           = lookup(network_interfaces.value, "ipv4_address_count", null)
+      ipv6_addresses               = lookup(network_interfaces.value, "ipv6_addresses", null) != null ? network_interfaces.value.ipv6_addresses : []
+      ipv6_address_count           = lookup(network_interfaces.value, "ipv6_address_count", null)
+      network_interface_id         = lookup(network_interfaces.value, "network_interface_id", null)
+      private_ip_address           = lookup(network_interfaces.value, "private_ip_address", null)
+      security_groups              = lookup(network_interfaces.value, "security_groups", null) != null ? network_interfaces.value.security_groups : []
+      subnet_id                    = lookup(network_interfaces.value, "subnet_id", null)
+    }
+  }
+
+  dynamic "placement" {
+    for_each = var.placement != null ? [var.placement] : []
+    content {
+      affinity          = lookup(placement.value, "affinity", null)
+      availability_zone = lookup(placement.value, "availability_zone", null)
+      group_name        = lookup(placement.value, "group_name", null)
+      host_id           = lookup(placement.value, "host_id", null)
+      spread_domain     = lookup(placement.value, "spread_domain", null)
+      tenancy           = lookup(placement.value, "tenancy", null)
+      partition_number  = lookup(placement.value, "partition_number", null)
+    }
+  }
+
+  dynamic "tag_specifications" {
+    for_each = var.tag_specifications
+    content {
+      resource_type = tag_specifications.value.resource_type
+      tags          = lookup(tag_specifications.value, "tags", var.tags_as_map)
```

Comment:
`var.tags_as_map` or the whole `local.tags`?

Reply:
I believe `var.tags_as_map` though I am not wondering if I should change this to 
```suggestion
      tags          = tag_specifications.value.tags
```
<img width="279" alt="image" src="https://user-images.githubusercontent.com/10913471/113793878-53befe80-9717-11eb-8b1a-a5a38c661d5d.png">

in our scenario, it is `tags_as_map` where its just a `key = value` pair, but I could also see where we could do a local variable to iterate over `local.tags` and provide a `local.tags_as_map` here, or just let users pick the tags they want to associate. the tags here are for specific resources as identified by `resource_type` so many not all tags are required, thoughts?
<img width="654" alt="image" src="https://user-images.githubusercontent.com/10913471/113793960-79e49e80-9717-11eb-8612-6db6f10553d9.png">

Reply:
I think the suggested `tag_specifications.value.tags` is better to allow granular control over tags per each resource type.

Reply:
fixed in https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/139/commits/50eb54f1fe485641bf3517661cbbc0dc7c37acd8